core/utils.go

```diff
@@ -301,3 +311,65 @@ func FileExists(path string) (bool, error) {
 
 	return false, err
 }
+
+func needsBom() bool {
+	// The reason why we do this is because notepad.exe on Windows determines the
+	// encoding of an "empty" text file by the locale, for example, GBK in China,
+	// while golang string only handles utf8 well. However, a text file with utf8
+	// BOM header is not considered "empty" on Windows, and the encoding will then
+	// be determined utf8 by notepad.exe, instead of GBK or other encodings.
+	// This could be enhanced in the future by doing this only when a non-utf8
+	// locale is in use, and possibly doing that for any OS, not just windows.
+	return runtime.GOOS == "windows"
+}
+
+// get body text with a text editor form env
+func getBodyTextWithEditor(ext string) (string, error) {
+	bodyTxtFile, err := os.CreateTemp(os.TempDir(), "changie-body-txt-*."+ext)
+	if err != nil {
+		return "", err
+	}
+
+	defer os.Remove(bodyTxtFile.Name())
+
+	if needsBom() {
+		if _, err = bodyTxtFile.Write(bom); err != nil {
+			return "", err
+		}
+	}
+
+	// close the file to prevent the editor unable to save file
+	if err = bodyTxtFile.Close(); err != nil {
+		return "", err
+	}
+
+	editor := os.Getenv("EDITOR")
+	if editor == "" {
+		return "", fmt.Errorf("'EDITOR' env variable not set")
```

Comment:
```suggestion
		return "", errors.New("'EDITOR' env variable not set")
```
Without any formatting you can use errors.New instead.